pkg/policy/run.go

```diff
@@ -89,52 +91,52 @@ type RunRequest struct {
 	RunCallback UpdateCallback
 }
 
-func Run(ctx context.Context, storage database.Storage, req *RunRequest) ([]*ExecutionResult, error) {
+func Run(ctx context.Context, storage database.Storage, req *RunRequest) ([]*ExecutionResult, diag.Diagnostics) {
 	results := make([]*ExecutionResult, 0)
+	var diags diag.Diagnostics
 	for _, p := range req.Policies {
 		log.Info().Str("policy", p.Name).Str("version", p.Version()).Str("subPath", p.SubPolicy()).Msg("preparing to run policy")
 		loadedPolicy, err := Load(ctx, req.Directory, p)
 		if err != nil {
-			return nil, err
+			return nil, diag.FromError(err, diag.INTERNAL)
 		}
 		log.Debug().Str("policy", p.Name).Str("version", p.Version()).Str("subPath", p.SubPolicy()).Msg("loaded policy successfully")
-		result, err := run(ctx, storage, &ExecuteRequest{
-			Policy:           loadedPolicy,
-			ProviderVersions: nil,
-			UpdateCallback:   req.RunCallback,
+		result, dd := run(ctx, storage, &ExecuteRequest{
+			Policy:         loadedPolicy,
+			UpdateCallback: req.RunCallback,
 		})
-
 		log.Info().Msg("policy execution finished")
-		if err != nil {
+		if dd.HasErrors() {
 			// this error means error in execution and not policy violation
 			// we should exit immediately as this is a non-recoverable error
 			// might mean schema is incorrect, provider version
 			log.Error().Err(err).Msg("policy execution finished with error")
-			return results, err
+			return results, diags.Add(dd)
 		}
 
+		diags = diags.Add(dd)
```

Comment:
maybe do diags = diags.Add(dd) on line 109, and do the if dd.HasErrors() afterwards, returning the diags itself if true